src/components/FlowPlayer/FlowPlayer.tsx

```diff
@@ -29,200 +35,215 @@ export interface FlowPlayerProps extends DefaultProps {
 	onPause?: () => void;
 	onEnded?: () => void;
 	onTimeUpdate?: (time: number) => void;
+	className?: string;
 }
 
-export const FlowPlayer: FunctionComponent<FlowPlayerProps> = ({
-	src,
-	poster,
-	title,
-	logo,
-	onInit,
-	start = 0,
-	end,
-	subtitles,
-	token,
-	dataPlayerId,
-	seekTime,
-	autoplay = true,
-	className,
-	onPlay = () => {},
-	onPause = () => {},
-	onEnded = () => {},
-	onTimeUpdate = () => {},
-}) => {
-	const videoContainerRef = useRef(null);
-	const videoPlayerRef: MutableRefObject<any | undefined> = useRef<any>();
-	const [lastSeekTime, setLastSeekTime] = useState<number | undefined>(undefined);
-
-	useEffect(() => {
-		return () => {
-			// Destroy handler
-			if (videoPlayerRef.current) {
-				videoPlayerRef.current.destroy();
-			}
+interface FlowPlayerState {
+	flowPlayerInstance: FlowplayerInstance | null;
+}
+
+export class FlowPlayer extends React.Component<FlowPlayerProps, FlowPlayerState> {
+	private videoContainerRef = createRef<HTMLDivElement>();
+
+	constructor(props: FlowPlayerProps) {
+		super(props);
+		this.state = {
+			flowPlayerInstance: null,
 		};
-	});
-
-	useEffect(() => {
-		if (typeof seekTime !== 'undefined' && seekTime !== lastSeekTime && seekTime !== 0) {
-			setLastSeekTime(seekTime);
-			// seekTime should be updated
-			if (videoPlayerRef.current) {
-				videoPlayerRef.current.currentTime = seekTime;
-			}
+	}
+
+	componentWillUnmount() {
```

Comment:
Je kan `unmount` en `shouldComponentUpdate` in principe beide uitvoeren via een `useEffect` of een custom React hook dus moest in principe niet overschakelen van de Function naar Class component. Maar voor mijn part is het veranderen naar een class-based component voor deze geen issue.

Reply:
hetgeen ik wil doen in niet mogelijk

ze raden aan om een memo functie te gebruiken met hooks: https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-shouldcomponentupdate

maar vanuit die memo compare functie kan ik niet aan het video reference element

Reply:
Ok, I get it. 👍